src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -2938,6 +2959,34 @@ public async Task TestEngineSupportViewsWithoutKeyFieldsInConfigForMsSQL()
             }
         }
 
+        [TestMethod, TestCategory(TestCategory.COSMOSDBNOSQL)]
+        public void ValidateGraphQLSchemaForCircularReference()
```

Comment:
need test summary. What is the circular reference?  which GraphQL type is processing expected to fail and throw exception? (character.moons or moons.character?)

Reply:
Circular Reference is a general term, I think it is not really required to explain explicitly what it really means. Anyway, I tried to put more information around.

Reply:
I didn't mean to explain the term circular reference. I meant for you to point out, in your sample `schema.gql` file, which properties are you expected to be detected as circular references.